flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -328,6 +329,7 @@ private void setBatchStateBackendAndTimerService(StreamGraph graph) {
         if (useStateBackend) {
             LOG.debug("Using BATCH execution state backend and timer service.");
             graph.setStateBackend(new BatchExecutionStateBackend());
+            graph.setCheckpointStorage(new BatchExecutionCheckpointStorage());
```

Comment:
Should there be a similar call for other cases (line 303)?

Reply:
This was in the follow up PR. This is the weird space where we are getting close the public API. I have backported `graph.setCheckpointStorage` on line 303ish but not setting the instance variable `checkpointStorage` that I have left for the follow up because it involves modifying `StreamExecutionEnvironment`.